libqtile/widget/google_calendar.py

```diff
@@ -106,32 +115,32 @@ def __init__(self, **config):
         self.default_foreground = self.foreground
 
     def cred_init(self):
-        # this is the main method for obtaining credentials
-        self.log.info('refreshing GC credentials')
-
-        # Set up a Flow object to be used for authentication.
-        FLOW = OAuth2WebServerFlow(
-            client_id='196949979762-5m3j4orcn9heesoh6td942gb2bph424q.'
-            'apps.googleusercontent.com',
-            client_secret='3H1-w_9gX4DFx3bC9c-whEBs',
-            scope='https://www.googleapis.com/auth/calendar',
-            user_agent='Qtile Google Calendar Widget/Version 0.3'
-        )
-
-        # storage is the location of our authentication credentials
-        storage = oauth2client.file.Storage(self.storage_file)
-
-        # get the credentials, and update if necessary
-        # this method will write the new creds back to disk if they are updated
+        """Gets valid user credentials from storage.
+
+        If nothing has been stored, or if the stored credentials are invalid,
+        the OAuth2 flow is completed to obtain the new credentials.
+        """
+        home_dir = os.path.expanduser('~')
+        credential_dir = os.path.join(home_dir, '.credentials')
+        if not os.path.exists(credential_dir):
+            os.makedirs(credential_dir)
+        credential_path = os.path.join(credential_dir,
+                                       'google-calendar-widget.json')
+        secret_path = os.path.join(credential_dir,
+                                   CLIENT_SECRET_FILE)
+
+        storage = oauth2client.file.Storage(credential_path)
         self.credentials = storage.get()
+        flow = client.flow_from_clientsecrets(secret_path, SCOPES)
+        flow.user_agent = APPLICATION_NAME
 
-        # if the credentials don't exist or are invalid, get new ones from FLOW
-        # FLOW must be run in a different thread or it blocks qtile
-        # when it tries to pop the authentication web page
         def get_from_flow(creds, storage):
-            self.credentials = run(FLOW, storage)
+            if flags:
+                self.credentials = tools.run_flow(flow, storage, flags)
+            else: # Needed only for compatibility with Python 2.6
```

Comment:
If I'm right, we don't support more  2.6 anymore, so we don't need more this lines